AppCenter/AppCenterTests/Util/MSMockKeychainUtil.m

```diff
@@ -68,14 +71,31 @@ + (NSString *_Nullable)deleteStringForKey:(NSString *)key withServiceName:(NSStr
   return value;
 }
 
-+ (NSString *_Nullable)stringForKey:(NSString *)key {
-  return [self stringForKey:key withServiceName:kMSDefaultServiceName];
++ (NSString *_Nullable)stringForKey:(NSString *)key statusCode:(OSStatus *)statusCode {
+  return [self stringForKey:key withServiceName:kMSDefaultServiceName statusCode:statusCode];
 }
 
-+ (NSString *_Nullable)stringForKey:(NSString *)key withServiceName:(NSString *)serviceName {
++ (NSString *_Nullable)stringForKey:(NSString *)key withServiceName:(NSString *)serviceName statusCode:(OSStatus *)statusCode {
+  OSStatus placeholderStatus = noErr;
+  if (statusCodes[serviceName] && statusCodes[serviceName][key]) {
+    placeholderStatus = [statusCodes[serviceName][key] intValue];
+  }
+  if (statusCode) {
+    *statusCode = placeholderStatus;
+  }
+  if (placeholderStatus != noErr) {
+    return nil;
+  }
   return stringsDictionary[serviceName][key];
 }
 
++ (void)mockStatusCode:(OSStatus)statusCode forKey:(NSString *)key {
```

Comment:
Why do you need this one? Can't you just remove all objects in `stopMocking ` method and use it?

Reply:
We specifically need to mock certain status codes - this isn't only used for cleanup

Reply:
but we can do that. still need to keep the method though.